### PR TITLE
Feature/assignment availability

### LINKED ISSUE #216: Assignment deadline & availability
Assignments should have deadlines and should stop being available after a certain date. Availability would then determine whether the user can still post entries.

#196 should also provide these details from LTI.

### PR DESCRIPTION
**Pull request by Airgetfrog**
## Description
Added assignment unlock, due, and lock dates. 
Adds relevant database and ui fields and lti functionality.
Adds background checks on get requests outside of unlocked period and entries posted after due date.
Updates ui in journal.

## Summary of changes
- Fixes #216 

## Screens
image
image
image

### CODE REVIEW

**Comment by Airgetfrog** [comment]
Everything appears to be in order.

File: src/vue/src/components/assignment/CreateAssignment.vue [hunk 1]
```diff
@@ -9,19 +9,37 @@
...
                 @content-update="form.assignmentDescription = $event"
                 :footer="false"
             />
+            <h2 class="field-heading">Points possible</h2>
+            <b-input
+                class="mb-2 mr-sm-2 mb-sm-0 multi-form theme-input"
+                v-model="form.pointsPossible"
+                placeholder="Points"
+                type="number"/>
+            <h2 class="field-heading">Unlock date</h2>
```
[the remark this answers is not in the document]
Comment (Airgetfrog) [reply src/vue/src/components/assignment/CreateAssignment.vue]: What should we do with the due date then? The natural progression for an assignment is unlock->due->locked, and all three in one line seems excessive, and I do feel they should be kept in order.

**Comment by Airgetfrog** [comment]
As for making these fields required, I think that depends on the canvas/lti implementation. Are they required in that as well?

File: src/vue/src/components/journal/JournalNonStudent.vue [hunk 2]
```diff
@@ -22,9 +22,23 @@
                         </b-card>
                     </div>
                 </div>
-                <b-card  v-else class="no-hover" :class="$root.getBorderClass($route.params.cID)">
```
[the remark this answers is not in the document]
Comment (Airgetfrog) [reply src/vue/src/components/journal/JournalNonStudent.vue]: These 2 (+format edit) are very slightly different but it seems like a good idea.

File: src/vue/src/components/assignment/CreateAssignment.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (mjvkeulen) [reply src/vue/src/components/assignment/CreateAssignment.vue]: I agree both ways can be argued. I just dislike the look of stacked inputs with spare space I guess

image

Upto you what you prefer.

**Pull request merged**